File: Backend/Microservices/Inventory/Eefa.Inventory.Domain/Entities/Role.cs
using Eefa.Common.Domain;
using System.Collections.Generic;

namespace Eefa.Inventory.Domain
{

    public partial class Role : DomainBaseEntity
    {
        public int? ParentId { get; set; }
        public string LevelCode { get; set; } = default!;
        public string Title { get; set; } = default!;
        public string UniqueName { get; set; } = default!;
        public string? Description { get; set; }        

    }
}

File: Frontend/angular-app/src/app/modules/admin/entities/role.ts

export class Role {
  public id!:number;
  public title!:string;
  public description!:string;
  public uniqueName!:string;
  public levelCode!:string;
  public parentId!:number;
  public copy !: string;
  public permissionsId: number[] = [];
}
